aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,201 @@
+"""EddsaJcs2022 cryptosuite."""
+
+from hashlib import sha256
+import canonicaljson
+
+from ....wallet.base import BaseWallet
+from ....wallet.keys.manager import MultikeyManager
+from ....utils.multiformats import multibase
+from ....core.profile import ProfileSession
+from ....resolver.did_resolver import DIDResolver
+from ..errors import PROBLEM_DETAILS
+
+
+class CryptosuiteError(Exception):
+    """Generic Cryptosuite Error."""
+
+
+class EddsaJcs2022:
+    """EddsaJcs2022 cryptosuite.
+
+    https://www.w3.org/TR/vc-di-eddsa/#eddsa-jcs-2022.
+    """
+
+    def __init__(self, *, session: ProfileSession):
+        """Create new EddsaJcs2022 Cryptosuite instance.
+
+        Args:
+            session: ProfileSession to use during crypto operations.
+
+        """
+        super().__init__()
+        self.session = session
+        self.wallet = session.inject(BaseWallet)
+        self.key_manager = MultikeyManager(session)
+
+    async def create_proof(self, unsecured_data_document: dict, options: dict):
+        """Create proof algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#create-proof-eddsa-jcs-2022.
+        """
+        proof = options.copy()
+
+        # Spec says to copy document context to the proof but it's unecessary IMO,
+        # commenting out for the time being...
+
+        # if '@context' in unsecured_data_document:
+        #     proof['@context'] = unsecured_data_document['@context']
+
+        proof_config = self.proof_configuration(proof)
+        transformed_data = self.transformation(unsecured_data_document, options)
+        hash_data = self.hashing(transformed_data, proof_config)
+        proof_bytes = await self.proof_serialization(hash_data, options)
+
+        proof["proofValue"] = multibase.encode(proof_bytes, "base58btc")
+
+        return proof
+
+    def proof_configuration(self, options: dict):
+        """Proof configuration algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-configuration-eddsa-jcs-2022.
+        """
+        proof_config = options.copy()
+
+        assert (
+            proof_config["type"] == "DataIntegrityProof"
+        ), 'Expected proof.type to be "DataIntegrityProof'
+        assert (
+            proof_config["cryptosuite"] == "eddsa-jcs-2022"
+        ), 'Expected proof.cryptosuite to be "eddsa-jcs-2022'
```

Comment:
Do we anticipate user input potentially failing these assertions? Or do these constitute a violation of the "interface contract?"

Reply:
It's here because the spec says to check for these, I tried to transform each spec line into a code line. It should never actually fail at this point since the DataIntegrity manager will parse the data before invoking this cryptosuite.

Reply:
> If proofConfig.type is not set to DataIntegrityProof or proofConfig.cryptosuite is not set to eddsa-jcs-2022, an error MUST be raised that SHOULD convey an error type of [PROOF_GENERATION_ERROR](https://www.w3.org/TR/vc-data-integrity/#PROOF_GENERATION_ERROR).